components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/deployment/config/deployer/ConfigDeployer.java

```diff
@@ -0,0 +1,252 @@
+        this.globalProperties = loadPropertiesFromFile(new File(globalPropertiesFilePath));
+        // Load sever conf property file
+        Properties serverConfigProperties = loadPropertiesFromFile(new File(serverConfPropertyPath));
+
+        Properties newServerConfigProperties = new Properties();
+        String isCertDeploymentEnabled = getValueOfKey(IS_CERT_DEPLOYMENT_ENABLED);
+        if (isCertDeploymentEnabled == null) {
+            isCertDeploymentEnabled = "true";
+        }
+        PropertyHolder.getInstance().setProperty(IS_CERT_DEPLOYMENT_ENABLED, isCertDeploymentEnabled);
+        if (serverConfigProperties.size() == 0 && configProperties.size() == 0 ) {
+            log.info("No configuration is used in the integration");
+        } else {
+            if (serverConfigProperties.size() > 0) {
+                for (Map.Entry<Object, Object> entry : serverConfigProperties.entrySet()) {
+                    String key = entry.getKey().toString();
+                    String type = entry.getValue().toString();
+                    if (configProperties.containsKey(key)) {
+                        type = configProperties.getProperty(key);
+                        configProperties.remove(key);
+                    }
+                    newServerConfigProperties.setProperty(key, type);
+                    processConfigProperties(key, type);
+                }
+            }
+            if (configProperties.size() > 0) {
+                for (Map.Entry<Object, Object> entry : configProperties.entrySet()) {
+                    String key = entry.getKey().toString();
+                    String type = entry.getValue().toString();
+                    newServerConfigProperties.setProperty(key, type);
+                    processConfigProperties(key, type);
+                }
+            }
+            writeServerConfFile(serverConfPropertyPath, newServerConfigProperties);
+        }
+    }
+
+    public void processConfigProperties(String key, String type) {
+        String value = getValueOfKey(key);
+        if (value != null) {
+            if (Objects.equals(type, "cert")) {
+                deployCert(key, value);
+            }
+            if (PropertyHolder.getInstance().hasKey(key)) {
+                String oldValue = PropertyHolder.getInstance().getPropertyValue(key);
+                if (!Objects.equals(oldValue, value)) {
+                    log.error(String.format("The value:[%s] of the key:[%s] has been " +
+                            "replaced with the new value:[%s].", oldValue, key, value));
+                }
+            }
+            PropertyHolder.getInstance().setProperty(key, value);
+        } else {
+            log.error(String.format("The value of the key:[%s] is not found.", key));
+        }
+    }
+
+    public void deployCert(String key, String path) {
+        if (ResolverFactory.getInstance().getResolver("$config:" + IS_CERT_DEPLOYMENT_ENABLED).resolve().
```

Comment:
As I remember we are throwing a resolverException when a key is not found. Should we handle that gracefully here ?

Reply:
This key is always on the map. This default value is `true`. https://github.com/wso2/product-micro-integrator/pull/3751/files#diff-f14d0dc665cb901616176017cd9f524f88e598a674bd11017be2fc857b50593fR122